narwhal/worker/src/batch_maker.rs

```diff
@@ -187,9 +204,57 @@ impl BatchMaker {
             .with_label_values(&[self.committee.epoch.to_string().as_str(), reason])
             .observe(size as f64);
 
+        // Note: the code below waits on the WAN to send messages to all others, the store
+        // to write and confirm the write, and then the primary to respond to being notified
+        // of a batch. While all this is going on the other batches are not being created and
+        // processed leading to an under utilization of resources.
+
         // Send the batch through the deliver channel for further processing.
-        if self.tx_message.send(batch).await.is_err() {
+        let (notify_done, done_sending) = tokio::sync::oneshot::channel();
+        if self
```

Comment:
Is it on purpose that we send the message through the delivery channel before persisting it to storage?

Reply:
The advantage of your code is that we can transmit and persist the batch in parallel

Reply:
Yes, indeed we want these to operate in parallel.